src/webots/engine/WbSimulationWorld.cpp

```diff
@@ -175,9 +175,11 @@ void WbSimulationWorld::step() {
     //              (if the real-time mode is enabled, of course)
     // mean *= 0.90;
 
-    if (mean > timeStep && mSleepRealTime > 0.0)
+    if (mean > timeStep && mSleepRealTime > 0.0) {
       mSleepRealTime -= 0.03 * timeStep;
-    else if (mean < timeStep)
+      if (mSleepRealTime < 0)
+        mSleepRealTime = 0.0;
```

Comment:
From a control point of view introducing the clamp here is not very good as it introduced a bias in the positive direction...
Personally I would have instead clamped this just before use.
```
mTimer->start(qMax(mSleepRealTime, 0));
```

Reply:
Yes, that's true. However, this case is extremely rare and we should also rename the `mSleepRealTime` member variable into a more complicated name to reflect that it's a proportional control parameter (and hence can be negative). For simplicity, I'd prefer to keep things like that.